sdk/nodejs/test/connect.spec.ts

```diff
@@ -37,4 +48,116 @@ describe("NodeJS sdk Connect", function () {
       assert(e instanceof GraphQLRequestError)
     }
   })
+
+  describe("Automatic Provisioned CLI Binary", function () {
+    let oldEnv: string
+    let oldCLIHost: string
+    let oldCLIScheme: string
+    let tempDir: string
+    let cacheDir: string
+
+    before(() => {
+      oldEnv = JSON.stringify(process.env)
+      oldCLIHost = bin._cliHost()
+      oldCLIScheme = bin._cliScheme()
```

Comment:
Not sure about the utility of those old* variables (from line 60-62). Same in the `after(()...`

Reply:
Oh I guess it's not reset process.env obj... 
I'm playing with Sinon.js that could stubs process.env values

Reply:
I'm still confused with `oldCLIHost` and `oldCLIScheme`

Reply:
@sipsma : I've just pushed a commit to remove them

Reply:
Hey sorry I didn't get back to this PR yesterday, but the reason for this is that CLI_HOST and CLI_SCHEME are set in the top level of the module, so changing them will affect any other tests. I think it technically doesn't hurt to leave them changed right now since no other test cases will hit the code paths that use them, but I'd rather leave them in since it could be very confusing in the future if someone is writing another test case and those vars have a different unexpected value. I went back to the original approach for now, but let me know what you think.

Reply:
We should find another way in a near futur, like stubbing with `sinon.js` for instance.

Reply:
Agreed, I tried a bunch of stuff like `rewire` but couldn't get it to work; I didn't think sinon would work because it would require we change our `Connect` method to accept a way of configuring all this, which means its configurable outside of tests too, which is what I was trying to avoid. Maybe it's just unavoidable though? Or I'm probably missing something obvious.